### PR TITLE
octopus: os/bluestore: Remove possibility of replay log and file inconsistency

### PR DESCRIPTION
**Pull request by ifed01**
backport tracker: https://tracker.ceph.com/issues/51128
backport of #41501
parent tracker: https://tracker.ceph.com/issues/50965

Signed-off-by: Igor Fedotov <ifedotov@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by ideepika — APPROVED** [review]

**Comment by ifed01** [comment]
Pacific backport is at: https://github.com/ceph/ceph/pull/42424

**Comment by ideepika** [comment]
thanks @ifed01 ! can you take a quick look at make check failure :
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/BlueFS.cc: In function 'int BlueFS::mkfs(uuid_d, const bluefs_layout_t&)' thread 7f77a3b035c0 time 2021-07-20T18:20:48.459321+0000
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/BlueFS.cc: 532: FAILED ceph_assert(r == 0)
 ceph version Development (no_version) octopus (stable)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x1a1) [0x7f77a5b8e19f]
 2: (()+0x159d43d) [0x7f77a5b8e43d]
 3: (BlueFS::mkfs(uuid_d, bluefs_layout_t const&)+0x559) [0x56395c8c0931]
 4: (BlueFS_test_tracker_50965_Test::TestBody()+0x4b0) [0x56395c85480a]
 5: (void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)+0x69) [0x56395c8a7a69]
 6: (void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)+0x5e) [0x56395c89ef33]
 7: (testing::Test::Run()+0xf2) [0x56395c877d1e]
 8: (testing::TestInfo::Run()+0x113) [0x56395c87871b]
```

File: src/test/objectstore/test_bluefs.cc [hunk 1]
```diff
@@ -840,6 +840,72 @@ TEST(BlueFS, test_replay_growth) {
   fs.umount();
 }
 
+TEST(BlueFS, test_tracker_50965) {
```
Comment (neha-ojha) [inline src/test/objectstore/test_bluefs.cc:843]: The failure is related to this test.

```
2021-07-20T18:20:48.454+0000 7f77a3b035c0 -1 bluefs _allocate failed to expand slow device to fit +0x400000
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/BlueFS.cc: In function 'int BlueFS::mkfs(uuid_d, const bluefs_layout_t&)' thread 7f77a3b035c0 time 2021-07-20T18:20:48.459321+0000
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/BlueFS.cc: 532: FAILED ceph_assert(r == 0)
 ceph version Development (no_version) octopus (stable)
 1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x1a1) [0x7f77a5b8e19f]
 2: (()+0x159d43d) [0x7f77a5b8e43d]
 3: (BlueFS::mkfs(uuid_d, bluefs_layout_t const&)+0x559) [0x56395c8c0931]
 4: (BlueFS_test_tracker_50965_Test::TestBody()+0x4b0) [0x56395c85480a]
 5: (void testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)+0x69) [0x56395c8a7a69]
 6: (void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)+0x5e) [0x56395c89ef33]
 7: (testing::Test::Run()+0xf2) [0x56395c877d1e]
 8: (testing::TestInfo::Run()+0x113) [0x56395c87871b]
 9: (testing::TestSuite::Run()+0x12d) [0x56395c878e73]
 10: (testing::internal::UnitTestImpl::RunAllTests()+0x41d) [0x56395c885161]
 11: (bool testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)+0x69) [0x56395c8a9060]
 12: (bool testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)+0x5e) [0x56395c8a0478]
 13: (testing::UnitTest::Run()+0xcd) [0x56395c8838ff]
 14: (RUN_ALL_TESTS()+0x15) [0x56395c858f8c]
 15: (main()+0x232) [0x56395c856124]
 16: (__libc_start_main()+0xf3) [0x7f77a40d90b3]
 17: (_start()+0x2e) [0x56395c843f9e]
```

**Review by neha-ojha — CHANGES_REQUESTED** [review]
@ifed01 can you please take a look at the make check failure

**Comment by ifed01** [comment]
> @ifed01 can you please take a look at the make check failure

hopefully fixed, lacked some (not needed in Pacific+) initialization stuff in bluefs UT...

**Pull request merged**